lib/Service/ImportService.php

```diff
@@ -274,9 +399,45 @@ private function getColumns(Row $firstRow, Row $secondRow): void {
 		$secondRowCellIterator = $secondRow->getCellIterator();
 		$titles = [];
 		$dataTypes = [];
+		$index = 0;
+		$countMatchingColumnsFromConfig = 0;
+		$countCreatedColumnsFromConfig = 0;
 		foreach ($cellIterator as $cell) {
 			if ($cell && $cell->getValue() !== null && $cell->getValue() !== '') {
-				$titles[] = $cell->getValue();
+				$title = $cell->getValue();
+
+				if (isset($this->columnsConfig[$index]) && $this->columnsConfig[$index]['action'] === 'exist' && $this->columnsConfig[$index]['existColumn']) {
+					$title = $this->columnsConfig[$index]['existColumn']['label'];
+					$countMatchingColumnsFromConfig++;
+				}
+				if (isset($this->columnsConfig[$index]) && $this->columnsConfig[$index]['action'] === 'new' && $this->createUnknownColumns) {
+					$column = $this->columnService->create(
+						$this->userId,
+						$this->tableId,
+						$this->viewId,
+						$this->columnsConfig[$index]['type'],
+						$this->columnsConfig[$index]['subtype'],
+						$this->columnsConfig[$index]['title'],
+						$this->columnsConfig[$index]['mandatory'] ?? false,
+						$this->columnsConfig[$index]['description'] ?? '',
+						$this->columnsConfig[$index]['textDefault'] ?? '',
+						$this->columnsConfig[$index]['textAllowedPattern'] ?? '',
+						$this->columnsConfig[$index]['textMaxLength'] ?? null,
+						$this->columnsConfig[$index]['numberPrefix'] ?? '',
+						$this->columnsConfig[$index]['numberSuffix'] ?? '',
+						$this->columnsConfig[$index]['numberDefault'] ?? null,
+						$this->columnsConfig[$index]['numberMin'] ?? null,
+						$this->columnsConfig[$index]['numberMax'] ?? null,
+						$this->columnsConfig[$index]['numberDecimals'] ?? 0,
+						$this->columnsConfig[$index]['selectionOptions'] ?? '',
+						$this->columnsConfig[$index]['selectionDefault'] ?? '',
+						$this->columnsConfig[$index]['datetimeDefault'] ?? '',
+						$this->columnsConfig[$index]['selectedViewIds'] ?? []
+					);
```

Comment:
Not originating from this PR but we should think about a better architecture then having this single method with tons of arguments. Something to track as a follow up issue.